test_agent/rust/src/utils.rs

```diff
@@ -0,0 +1,552 @@
+            .and_then(|authority| authority.get("name"))
+        {
+            Some(_authority_name) => _authority_name.as_str(),
+            None => Some("default"),
+        };
+
+        let _authority_number_ip = match value
+            .get("authority")
+            .and_then(|authority| authority.get("number"))
+            .and_then(|number| number.get("Ip"))
+        {
+            Some(_authority_number_ip) => _authority_number_ip.to_string().as_bytes().to_vec(),
+            None => {
+                let default: Vec<u8> = vec![0];
+                default
+            }
+        };
+        let _authority_number_id = match value
+            .get("authority")
+            .and_then(|authority| authority.get("number"))
+            .and_then(|number| number.get("Id"))
+        {
+            Some(_authority_number_id) => _authority_number_id.to_string().as_bytes().to_vec(),
+            None => {
+                let default: Vec<u8> = vec![0];
+                default
+            }
+        };
+
+        let mut _authority = UAuthority::new();
+        if!( _authority_name.clone() == Some("default")){
+            _authority.name = _authority_name.map(|s| s.to_string());
+        }
+        if!(_authority_number_id.clone() == vec![0]){
+            _authority.set_id(_authority_number_id.clone());
+        }
+        if!(_authority_number_ip.clone() == vec![0]){
+            _authority.set_ip(_authority_number_ip.clone());
+        }
+
+        //update entity
+        let _entity_name = match value.get("entity").and_then(|entity| entity.get("name")) {
+            Some(_entity_name) => _entity_name.as_str(),
+            None => Some("default"),
+        };
+        let _entity_id = match value.get("entity").and_then(|entity| entity.get("id")) {
+            Some(_entity_id) => _entity_id
+                .clone()
+                .as_str()
+                .expect("not a string")
+                .parse::<u32>()
+                .expect("issue in converting to u32"),
+            None => 0,
+        };
+        let _entity_version_major = match value
+            .get("entity")
+            .and_then(|entity| entity.get("version_major"))
+        {
+            Some(_entity_version_major) => _entity_version_major
+                .clone()
+                .as_str()
+                .expect("not a string")
+                .parse::<u32>()
+                .expect("issue in converting to u32"),
+
+            None => 0,
+        };
+        let _entity_version_minor = match value
+            .get("entity")
+            .and_then(|entity| entity.get("version_minor"))
+        {
+            Some(_entity_version_minor) => _entity_version_minor
+                .clone()
+                .as_str()
+                .expect("not a string")
+                .parse::<u32>()
+                .expect("issue in converting to u32"),
+            None => 0,
+        };
+        let _entity_special_fields = SpecialFields::default();
```

Comment:
Like shown in [this comment](https://github.com/eclipse-uprotocol/up-tck/pull/23/files#r1566166247) we can use a combination of the `Default` trait and the `..` syntax together in a powerful combination that lets us copy in the default values of structs, including the `SpecialFields` content (which is used only by the Rust protobuf crate anyways).

But it can be used for other kinds of interesting things too.

For example, say you have a `UResource` which has only a name and method, you can do this:
```rust
        let my_cool_resource = UResource {
            name: "very_cool_resource".to_string(),
            instance: Some("my_very_cool_instance".to_string()),
            id: Some(111),
            ..Default::default()
        };
```
and the message portion will be just set to `None`.